tests/test_cli.py

```diff
@@ -0,0 +1,164 @@
+
+    set_setting("temperature", "0.5")
+    assert show("temperature") == "0.5", "Temperature not properly set"
+
+    clear("temperature")
+    assert show("temperature") is None, "Temperature not properly cleared"
+
+    # set a nested variable
+    set_setting("prompts.qa", "Answer my question!")
+    assert show("prompts.qa") == "Answer my question!", "Prompt not properly set"
+    clear("prompts.qa")
+    assert show("prompts.qa") is None, "Prompt not properly cleared"
+
+    # creates a new index/file
+    si = SearchIndex(index_directory=agent_index_dir)
+    await si.init_directory()
+
+    clear("pqa_index", index=True)
+
+    assert show("indexes") == [], "Index not properly cleared"
+
+
+def test_cli_ask(agent_index_dir):
+    set_setting("consider_sources", "10")
+    set_setting("max_sources", "2")
+    set_setting("embedding", "sparse")
+    set_setting("agent_tools.search_count", "1")
+    answer = ask(
+        "How can you use XAI for chemical property prediction?",
+        directory=str(Path(__file__).parent),
+        index_directory=agent_index_dir,
+    )
+    assert isinstance(answer, AnswerResponse), "Answer not properly returned"
+    assert (
+        "I cannot answer" not in answer.answer.answer
+    ), "An answer should be generated."
+    assert len(answer.answer.context) >= 1, "No contexts were found."
+    answers = search_query(
+        "How can you use XAI for chemical property prediction?",
+        index_directory=agent_index_dir,
+    )
+    answer = answers[0][0]
+    assert isinstance(answer, AnswerResponse), "Answer not properly returned"
+    assert (
+        "I cannot answer" not in answer.answer.answer
+    ), "An answer should be generated."
+    assert len(answer.answer.context) >= 1, "No contexts were found."
+    assert len(show("answers")) == 1, "An answer should be returned"
+
+
+def test_cli_index(agent_index_dir, caplog):
+
+    _ = build_index(
+        directory=str(Path(__file__).parent), index_directory=agent_index_dir
+    )
+
+    caplog.clear()
+
+    _ = ask(
+        "How can you use XAI for chemical property prediction?",
+        directory=str(Path(__file__).parent),
+        index_directory=agent_index_dir,
+    )
+
+    # ensure we have no indexing logs after starting the search
+    for record in caplog.records:
+        if "Metadata not found" in record.msg:
+            raise AssertionError(
+                "Indexing logs should not be present after search starts"
+            )
+
+    caplog.clear()
+
+    # running again should not trigger any indexing
+    _ = build_index(
+        directory=str(Path(__file__).parent), index_directory=agent_index_dir
+    )
+
+    assert not caplog.records, "Indexing should not be triggered again"
+
+    # now we want to change the settings
+    set_setting("embedding", "sparse")
+
+    # running again should now re-trigger an indexing
+    _ = build_index(
+        directory=str(Path(__file__).parent), index_directory=agent_index_dir
```

Comment:
Can we make `build_index` support `os.PathLike` so we don't need these `str` casts